core/metadata.js

```diff
@@ -240,16 +238,25 @@ function sameFile (one /*: Metadata */, two /*: Metadata */) {
   }
 }
 
+// Return true if the metadata of the two files are the same,
+// ignoring revision
 function sameFileIgnoreRev (one /*: Metadata */, two /*: Metadata */) {
   const {path} = two
   let fields = ['_id', 'docType', 'md5sum', 'remote._id', 'tags', 'size', 'trashed', 'ino']
   one = _.merge({executable: !!one.executable}, pick(one, fields))
   two = _.merge({executable: !!two.executable}, pick(two, fields))
   const same = isEqual(one, two)
-  if (!same) log.trace({path, diff: {one, two}})
+  if (!same) log.trace({path, one, two, diff: explainDiff(one, two)})
   return same
 }
 
+function explainDiff (one /*: Metadata */, two /*: Metadata */) {
```

Comment:
What about having a `diff()` function that returns an object with differences if any or nothing otherwise?
It would ensure there is no differences between the way we explain the comparison and the way we actually do it.
Also we wouldn't have to compare fields twice.

Reply:
Since this may be quite a critical change, maybe we should unit test the `core/metadata.same*()` functions before doing so... wdyt?